src/App.tsx

```diff
@@ -17,11 +17,12 @@ export const App: React.FC = () => {
   const timeoutRef = useRef<NodeJS.Timeout | null>(null);
   const [tempTodo, setTempTodo] = useState<Todo | null>(null);
   const [inputText, setInputText] = useState('');
-  const [deletedIds, setDeletedIds] = useState<number[]>([]);
   const [inputDisabled, setInputDisabled] = useState(false);
-  const [updatingIds, setUpdatingIds] = useState<number[]>([]);
+  const [loadingIds, setLoadingIds] = useState<{
+    updating: number[];
+    deleting: number[];
```

Comment:
you don't need two arrays, the is only one trigger at a time for loading todos. so you can use one array